controllers/keda/clustertriggerauthentication_controller.go

```diff
@@ -66,7 +66,9 @@ func (r *ClusterTriggerAuthenticationReconciler) Reconcile(ctx context.Context,
 		if errors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
+		msg := "Failed to get ClusterTriggerAuthentication"
 		reqLogger.Error(err, "Failed to get ClusterTriggerAuthentication")
+		r.EventRecorder.Event(clusterTriggerAuthentication, corev1.EventTypeNormal, eventreason.ClusterTriggerAuthenticationFailed, msg)
```

Comment:
```suggestion
		msg := "Failed to get ClusterTriggerAuthentication"
		reqLogger.Error(err, msg)
		r.EventRecorder.Event(clusterTriggerAuthentication, corev1.EventTypeNormal, eventreason.ClusterTriggerAuthenticationFailed, msg)
```

Reply:
Will this work? I mean, we have failed getting the resource, will we be able to update it or we'll receive a panic?

Reply:
Sorry, it will not work. An error log about 'can't reference a nil object' will occur. Do you have any ideas on how to improve this situation, or should we leave it as is?

Reply:
In this specific case, I'd keep it as it was. We cannot update something that we can't find xD